src/System.ServiceProcess.ServiceController/tests/System.ServiceProcess.ServiceController.TestService/TestService.cs

```diff
@@ -76,26 +82,11 @@ protected override void OnStop()
         {
             WriteLog(nameof(OnStop));
             base.OnStop();
-
-            if (_log != null)
-            {
-                _log.Dispose();
-                _log = null;
-            }
         }
 
-        private StreamWriter _log;
-
         private void WriteLog(string msg)
         {
-            if (_log == null)
-            {
-                string path = System.Reflection.Assembly.GetEntryAssembly().Location + "." + ServiceName + ".log";
-                _log = new StreamWriter(path);
-                _log.AutoFlush = true;
-;            }
-
-            _log.WriteLine(msg);
+             File.AppendAllText(GetLogPath(ServiceName), msg + Environment.NewLine);
```

Comment:
This is likely going to cause more race conditions with the file being opened and closed multiple times test method.